### DISCUSSION TITLE
The results vary when using different values of NLIST under the IVF_SQ8 index.

### DISCUSSION THREAD

**Discussion by pincutionhot** [milvus-io/milvus#26510] [Q&A and General discussion]
During my usage of Milvus, I noticed that when I created an index type IVF_SQ8 with nlist set to either 1024 or 2048, the results differed when comparing the same source and target data with different nlist values. Could you please help me understand why? Thank you.

**Comment by xiaofan-luan** [comment]
I think this expect behaviour.
the highr nlist, the recall is usually lower with same nprobe.

**Reply by pincutionhot** [reply]
Thank you for your response. My current issue is as follows: using vector a to search for vector b in the library. Despite different nlist settings, b appears in the query results, but the obtained results are not consistent. I would like to understand the underlying principles and the range of deviations involved.

**Reply by pincutionhot** [reply]
Because my algorithm not only focuses on recall rate but also aims to obtain IP results. Additionally, due to the large size of the collection, I have chosen IVF_SQ8

**Reply by xiaofan-luan** [reply]
to understand more about idnex, maybe faiss document is good to refer
https://github.com/facebookresearch/faiss/wiki/The-index-factory/55af8daecf02655359e9b9629fc65643fac42d86

**Reply by yhmo** [reply]
For IVF_SQ8 index, the index engine performs a scalar quantizer for each vector, thus the distance value is a bit different from the actual distance. Then the order of results might be different with different nlist and nprobe values.

**Reply by pincutionhot** [reply]
Thank you very much for your response. Its very helpful to me.  Is there faster way to obtain the actual IP results when using the IVF_SQ8 index? Currently, my workaround is to query the results from the candidates using the MilvusServiceClient.query interface then calculate the IP results again. Do you have any better suggestions?

**Reply by xiaofan-luan** [reply]
If you need actual vector, I suggest to use HNSW or IVFFLAT for performance reason.
